tests/cmdhandler/camera_system_tests.cc

```diff
@@ -0,0 +1,140 @@
+//
+// Created by tim on 3/31/21.
+//
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "../../src/cmdhandler/command_handler.h"
+
+using ::testing::HasSubstr;
+
+StateVariables testing_state_camera_sys = StateVariables();
+std::vector<std::string> input_camera_sys;
+
+TEST(CameraSystemSuite, Sets_Variables)
+{
+    input_camera_sys.clear();
+    input_camera_sys.insert(input_camera_sys.end(), {"set", "camera", "source", "testingurl"});
+    std::string response = command_handler::do_command(input_camera_sys,testing_state_camera_sys);
+
+    EXPECT_THAT(response, HasSubstr("source set to"));
+    ASSERT_EQ(testing_state_camera_sys.camera.source, "testingurl");
+
+    input_camera_sys.clear();
+    input_camera_sys.insert(input_camera_sys.end(), {"set", "camera", "marker_dictionary", "6"});
+    response = command_handler::do_command(input_camera_sys,testing_state_camera_sys);
+
+    EXPECT_THAT(response, HasSubstr("'marker_dictionary' variable set"));
+    ASSERT_EQ(testing_state_camera_sys.camera.marker_dictionary, 6);
+
+    input_camera_sys.clear();
+    input_camera_sys.insert(input_camera_sys.end(), {"set", "camera", "camera_options", "testing", "false"});
+    response = command_handler::do_command(input_camera_sys,testing_state_camera_sys);
+
+    EXPECT_THAT(response, HasSubstr("'testing' set to"));
+    ASSERT_EQ(testing_state_camera_sys.camera.camera_options.at("testing"), false);
+
+    input_camera_sys.clear();
+    input_camera_sys.insert(input_camera_sys.end(), {"set", "camera", "camera_matrix", "1,2,3,4,5,6,7,8,9"});
+    response = command_handler::do_command(input_camera_sys,testing_state_camera_sys);
+
+    EXPECT_THAT(response, HasSubstr("'camera_matrix' variable set"));
+    ASSERT_EQ(testing_state_camera_sys.camera.camera_matrix.at<double>(0), 1);
```

Comment:
gtest has `ASSERT_FLOAT_EQ` and `ASSERT_DOUBLE_EQ` macros that should be used for floating point comparisons

https://developer.ibm.com/technologies/systems/articles/au-googletestingframework/#list9